docs(typescript): add typescript directory (#24)
yseop/codequality: typescript/README.adoc

= Yseop code style documentation

:experimental:
:toc:
:toclevels: 3

This document's goal is to help configure a TypeScript environment.

== VSCode

A default `settings.json` file is available here: link:vscode/settings.json[`vscode/settings.json`].
It is correctly configured to use ESLint.

=== Useful extensions

> A huge amount of extensions exists on the Marketplace and the following could useful for you:

Reviewer: I kinda love the fact that you say “huge” and stuff but then only list one xD

`SonarLint`::
    IDE extension that helps you detect and fix quality issues as you write code in JavaScript, TypeScript, Python and PHP.

=== Extensions configuration

In Visual Studio Code, settings can be saved on a `local` scope or in the `workspace` (`.vscode/` folder, shared via Git).

==== SonarLint specific configuration

If you want to use the SonarLint extension, first, you need to generate an access token on SonarCloud (https://sonarcloud.io/account/security/).
It's used by SonarLint in order to fetch the configuration used on the project. +
Now you will have to update your `User Settings`:

> * Open the `Command Palette` using kbd:[CTRL+SHIFT+P]

Reviewer: ```suggestion
* Open the “Command Palette” using kbd:[Ctrl+Shift+P]
```

* Search `Preferences: Open Settings (JSON)`

> * Press kbd:[ENTER]

Reviewer: ```suggestion
* Press kbd:[Enter]
```

> * Add the following configuration and save
> 

Reviewer: Link snippet to the list item it is relevant for with a plus-line thingy.

```suggestion
* Add the following configuration and save:
+
```

```JSON
"sonarlint.ls.javaHome": "/usr/lib/jvm/java", <1>
"sonarlint.connectedMode.servers": [
    {
        "serverId": "sonarcloud.io",
        "serverUrl": "https://sonarcloud.io",
        "organizationKey": "yseop",
        "token": "xxx"
    }
]

> ```

Reviewer: See sugg above.

```suggestion
----
```

= Yseop code style documentation

:experimental:
:toc:
:toclevels: 3

This document's goal is to help configure a TypeScript environment.

== VSCode

A default `settings.json` file is available here: link:vscode/settings.json[`vscode/settings.json`].
It is correctly configured to use ESLint.

=== Useful extensions

A huge amount of extensions exists on the Marketplace and the following could useful for you:

`SonarLint`::
    IDE extension that helps you detect and fix quality issues as you write code in JavaScript, TypeScript, Python and PHP.

=== Extensions configuration

In Visual Studio Code, settings can be saved on a `local` scope or in the `workspace` (`.vscode/` folder, shared via Git).

==== SonarLint specific configuration

If you want to use the SonarLint extension, first, you need to generate an access token on SonarCloud (https://sonarcloud.io/account/security/).
It's used by SonarLint in order to fetch the configuration used on the project. +
Now you will have to update your `User Settings`:

* Open the "Command Palette" using kbd:[Ctrl+Shift+P]
* Search `Preferences: Open Settings (JSON)`
* Press kbd:[Enter]

> * Add the following configuration and save:

Reviewer: Woops these are consecutive steps so numbered list.

Reviewer: ```suggestion
. Open the "Command Palette" using kbd:[Ctrl+Shift+P]
. Search `Preferences: Open Settings (JSON)`
. Press kbd:[Enter]
. Add the following configuration and save:
```

= Yseop code style documentation

:experimental:

> :toc2:
> :sectnums:
> :revnumber: 1.0
> :icons: font
> :source-highlighter: coderay
> :sectanchors:
> :imagesdir: readme-img

Reviewer: Old and overkill IMO. Take example on more simple pages, such as the Bash part around here.
And no use setting up an “images dir” if there are no images to begin with.
Crap though, I notice via this that I used a different images dir name in the Bash part with respect to the one in the Java part. Will have to fix that myself for consistency’s sake.

```suggestion
:toc:
:toclevels: 3
```

> This document's goal is to help configure a typescript environment.

Reviewer: Please use the real names of languages and products and stuff, including capitalization.
https://en.wikipedia.org/wiki/TypeScript

==== VSCode

> A default settings.json file is available here `{PATH_TO_CODEQUALITY_REPOSITORY}/typescript/vscode/settings.json`. It is correctly configured to use ESLint.

Reviewer: * https://github.com/yseop/yseop-doc/blob/master/YSEOP_WRITER_GUIDE.adoc#inline-code-filenames
* https://github.com/yseop/yseop-doc/blob/master/YSEOP_WRITER_GUIDE.adoc#escaping-curly-braces
* https://github.com/yseop/yseop-doc/blob/master/YSEOP_WRITER_GUIDE.adoc#general-principles

And you could probably just link to the file (with a relative path), to make it easier to open it directly via GitHub and stuff.

```suggestion
A default `settings.json` file is available here: link:vscode/settings.json[`vscode/settings.json`].
It is correctly configured to use ESLint.
```

> 

Reviewer: ![image](https://user-images.githubusercontent.com/36635838/186122545-e9b537b3-f5ef-48b1-8cba-378dacc95f4d.png)
https://github.com/yseop/yseop-doc/blob/master/YSEOP_WRITER_GUIDE.adoc#headings-and-blank-lines

=== Recommended extensions

> The **VSCode workspace** references the recommended extensions to develop on this project. +

Reviewer: * Overkill double symbol. This is not Markdown.

https://github.com/yseop/yseop-doc/blob/master/YSEOP_WRITER_GUIDE.adoc#unconstrained-formatting

* Irrelevant forced linebreak.

https://github.com/yseop/yseop-doc/blob/master/YSEOP_WRITER_GUIDE.adoc#forced_linebreaks

```suggestion
The *VSCode workspace* references the recommended extensions to develop on this project.

```

> Install these extensions from VSC Marketplace.

Reviewer: Which VSCode workspace are you referring to? I see no such thing is that repository.

Author: Tu as raison, c'est un mauvais copier/coller (j'ai repris la doc que l'on avait sur le repo jslib)

=== Other extensions

> A huge amount of extensions exists on the Marketplace and these following could useful for you:

Reviewer: ```suggestion
A huge amount of extensions exists on the Marketplace and the following could useful for you:
```

`SonarLint`::
    IDE extension that helps you detect and fix quality issues as you write code in JavaScript, TypeScript, Python and PHP.

=== Extensions configuration

> In Visual Studio Code, settings can be saved on a `local` scope or in the `workspace` (`.vscode` folder, shared via `Git`). +

Reviewer: * `git` (the executable) or Git, without backticks (the project / product / whatever). Mixing those is confusing, kinda like pointing to a program that does not exist.

* See below from the guide.

![image](https://user-images.githubusercontent.com/36635838/186124959-bc33f584-5945-466e-98ca-6432ae32e2f0.png)

* Another irrelevant forced break (see link in previous comments).

```suggestion
In Visual Studio Code, settings can be saved on a `local` scope or in the `workspace` (`.vscode/` folder, shared via Git).

```

> For this project, needed extensions are already configured in the workspace settings (link:[../aa_js-commons.code-workspace]).

Reviewer: You keep saying “this project” and stuff but you kinda seem to have imported all this content from somewhere else and refer to some other project. I’m confused.

Author: yes, this is the doc from jslib that I centralized here, as you are right it could be useful to other people

==== SonarLint specific configuration

If you want to use the SonarLint extension, first, you need to generate an access token on SonarCloud (https://sonarcloud.io/account/security/).
It's used by SonarLint in order to fetch the configuration used on the project. +
Now you will have to update your `User Settings`:

> - Open the `Command Palette` using kbd:[CTRL+SHIFT+P]

Reviewer: Guide.

![image](https://user-images.githubusercontent.com/36635838/186125767-5222a4bc-7c3e-4132-9401-240b6ed5178e.png)

And this use of backticks for things such as “Command Palette” or “User Settings” is kinda dubious. Many of these cases fall in one of the two following categories:

1. See below (guide, but I’m not inventing anything here, just doing what is done in books and whatnot):

![image](https://user-images.githubusercontent.com/36635838/186126188-ffc03c29-0d92-4a22-b95d-9b83faf6b6e7.png)

2. UI elements: https://github.com/yseop/yseop-doc/blob/master/YSEOP_WRITER_GUIDE.adoc#references-to-user-interface-elements

- Search `Preferences: Open Settings (JSON)`
- Press kbd:[ENTER]
- Add the following configuration and save

> ```JSON

Reviewer: No Markdown-style code snippets please. They even trigger warnings in IJ by default, and make it hard to add configuration attributes and stuff.
https://github.com/yseop/yseop-doc/blob/master/YSEOP_WRITER_GUIDE.adoc#code-blocks

Reviewer: Boudouuuu ignored comment from before?

![pout4](https://user-images.githubusercontent.com/36635838/186185737-0ebe502d-2d2d-428f-97a8-820a6c05d443.jpg)

```suggestion
[source, json]
----
```

> // The location of your Java home

Reviewer: Use callouts when you can instead of comments, especially in JSON since comments are not supposed to be a thing in JSON to begin with.

https://github.com/yseop/yseop-doc/blob/master/YSEOP_WRITER_GUIDE.adoc#callouts

Author: ok, but it is a lot easier to read like that, no ?
this is just a quick internal doc initialization :)

Reviewer: Not necessarily easier to read when you’re used to proper stuff. And callouts tend to point more precisely to a specific part. And you call that a quick initialization but I highly doubt anyone is going to change much afterwards or clean anything. And anyway we should take on (and keep) good habits.
And not my fault if base JSON has no comment support. 😅

"sonarlint.ls.javaHome": "/usr/lib/jvm/java",
"sonarlint.connectedMode.servers": [
    {
        "serverId": "sonarcloud.io",
        "serverUrl": "https://sonarcloud.io",
        "organizationKey": "yseop",
        "token": "xxx"
    }

> ],

Reviewer: Strange to end with a comma like this. Either remove it or write `[…]` below it if this is supposed to be an excerpt rather than a full representative file or whatever.
